app/models/manageiq/providers/cloud_manager/template.rb

```diff
@@ -55,6 +55,45 @@ def self.create_image(ems_id, options)
     klass.raw_create_image(ext_management_system, options)
   end
 
+  # Queue import of image with an id 'options[:image_id]' from a provider
+  # with an id 'options[:provider_id]' into current provider
+  #
+  def self.import_image_queue(userid, ext_management_system, options = {})
+    task_opts = {
+      :action => "Importing Cloud Template for user #{userid}",
+      :userid => userid
+    }
+
+    queue_opts = {
+      :class_name  => name,
+      :method_name => 'import_image',
+      :role        => 'ems_operations',
+      :zone        => ext_management_system.my_zone,
+      :queue_name  => ext_management_system.queue_name_for_ems_operations,
+      :args        => [ext_management_system.id, options]
+    }
+
+    MiqTask.generic_action_with_callback(task_opts, queue_opts)
+  end
+
+  def self.raw_import_image(_ext_management_system, _options = {})
+    raise NotImplementedError, "raw_import_image must be implemented in a subclass"
+  end
+
+  def validate_import_image(_ext_management_system, _options = {})
+    validate_unsupported(_("Import Image Operation"))
+  end
```

Comment:
Prefer to use supports_feature over `validate_*` as this is an older style we're trying to move away from.

You can simply add `supports_not :import_image` to this class and in your provider add `supports :import_image`  (note this is much simpler after https://github.com/ManageIQ/manageiq/pull/21046 as you don't have to add anything to the `QUERYABLE_FEATURES` hash)

Reply:
Does this work with static methods? I tried that one out by extending the _SupportsFeatureMixin.rb_ and then adding _supports :import_image_ to the implementing class without any luck.

Reply:
I will try it out again now.

Reply:
Tried it out again. It didn't work in my case. I think it has to do with the fact that the method is static. Do you have an idea?

Reply:
Can you extrapolate on what doesn't work?  I just tried:
```
--- a/app/models/manageiq/providers/cloud_manager/template.rb
+++ b/app/models/manageiq/providers/cloud_manager/template.rb
@@ -3,6 +3,8 @@ class ManageIQ::Providers::CloudManager::Template < ::MiqTemplate
 
   virtual_column :image?, :type => :boolean
 
+  supports_not :import_image
```
```
--- a/app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/template.rb
+++ b/app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/template.rb
@@ -1,4 +1,5 @@
 class ManageIQ::Providers::IbmCloud::PowerVirtualServers::CloudManager::Template < ManageIQ::Providers::CloudManager::Template
+  supports :import_image
```
```
>> ManageIQ::Providers::CloudManager::Template.supports?(:import_image)
=> false
>> ManageIQ::Providers::IbmCloud::PowerVirtualServers::CloudManager::Template.supports?(:import_image)
=> true
```

Reply:
Well, I get the same values as you. However, if I comment out

`supports :import_image`

in _ManageIQ::Providers::IbmCloud::PowerVirtualServers::CloudManager::Template_

The method  _self.raw_import_image_ still gets executed.

Reply:
This has to be checked in the API call, there are lots of examples of checking `supports?()` there but let me know if you run into issues

Reply:
Okay, I assumed there must be some dynamic code in MIQ already that does just that. Get it, then I'll add the check to my API endpoint.

Reply:
I will then remove _validate_import_image_ method.